### ISSUE TITLE
Lint/ParenthesesAsGroupedExpression false positive with defined?

### ISSUE THREAD

**Issue by MatzFan** [rubocop/rubocop#15178]
### Before submitting

- [x] I have searched the existing issues and this hasn't been reported yet.
- [x] I am running the latest stable version of RuboCop.

### Issue type

False positive

### Cop name

Lint/ParenthesesAsGroupedExpression

### Steps to reproduce the problem

Create file `test_foo.rb`:
```ruby
# frozen_string_literal: true

require 'minitest/autorun'

# test
class FooTest < Minitest::Test
  def test_foo
    assert defined?(FooTest)
  end
end
```

`$ rubocop test_foo.rb`

### Expected behavior

No offenses should be reported

### Actual behavior

```
test_foo.rb:8:11: W: [Correctable] Lint/ParenthesesAsGroupedExpression: defined?(FooTest) interpreted as grouped expression.
    assert defined?(FooTest)
          ^
```

### RuboCop configuration

```yaml
N/A
```

### RuboCop version

```shell
1.86.2 (using Parser 3.3.11.1, rubocop-ast 1.49.1, analyzing as Ruby 2.7, running on ruby 4.0.5) [x86_64-linux]
```

**Comment by koic** [comment]
This issue is a duplicate of #15168 and resolved by #15169. It will be fixed in the next release.
